_build/stac.hbs

```diff
@@ -34,6 +34,11 @@
       "href": "{{rootUrl}}stac/{{Slug}}.json",
       "rel": "self"
     },
+    {
+      "href": {{{find WMTS true 'href'}}},
+      "rel": "wmts",
+      "wmts:layer": "{{{find WMTS true 'layer_id'}}}"
+    },

```

Comment:
I changed this added section to the code below.

```hbs
    {{#if WMTS }}
    {
      "href": "{{{find WMTS 'href' 'href'}}}",
      "rel": "wmts",
      "wmts:layer": "{{{find WMTS 'layer_id' 'layer_id'}}}"
    },
    {{/if}}
```

Explanation:

- `{{#if WMTS }}` checks if the WMTS part is present in the yaml file - without that, the `build` and `serve` script were returning an error
```
[15:09:07] 'stacGenerate' errored after 1.49 s
[15:09:07] TypeError in plugin "gulp-hb"
Message:
    Cannot read property 'length' of undefined
Details:
    fileName: /home/zcernigoj/Documents/dev/EDCPublicCollectionsRepos/main/public-collections/_build/stac.hbs
    domainEmitter: [object Object]
    domainThrown: false

Stack:
TypeError: Cannot read property 'length' of undefined
    at Object.find (/home/zcernigoj/Documents/dev/EDCPublicCollectionsRepos/main/public-collections/gulpfile.js:525:29)
    ...
```

- `"href": "{{{find WMTS 'href' 'href'}}}",` and `"wmts:layer": "{{{find WMTS 'layer_id' 'layer_id'}}}"`
  - wrapped in `"` as in the other instance of `find` above (for `datasource_type`)
  - fixed `find` statement, so it correctly finds the correct data and names it correctly in the yaml file

Reply:
Can you check if my current changes produce the correct results --> I don't think the  `resultField` should be present as field name in the stac - json as the name in the stac jsons is written before the find statement. The testament should only find the correct value and parse it here.

Reply:
Ah thanks, I was too quick, didn't check thoroughly and gave the wrong explanation of the `resultField`.

- `conditionalField` is the field for which the `find` checks if it's present
- `resultField` is the field that is returned 

My example above is correct to achieve what you describe, only the explanation was wrong. Sorry for the confusion.

example for `landsat-8-l1.yaml`:

```json
    {
      "href": "https://services.sentinel-hub.com/ogc/wmts/fc306b9b-4448-41d2-922d-6db03b26610e",
      "rel": "wmts",
      "wmts:layer": "TRUE-COLOR"
    },
```

Reply:
Also noticed that in this changed version, there are two `"` at the end of the `href` line, which get transferred to the `json` and the two `"` break it.

Reply:
Ah I understand now, thanks for the additional explanation. Fixed